pkg/kubevirt/apis/provider_spec.go

```diff
@@ -16,21 +16,22 @@ package api
 
 import (
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	kubevirtv1 "kubevirt.io/client-go/api/v1"
 )
 
 // KubeVirtProviderSpec is the spec to be used while parsing the calls.
 type KubeVirtProviderSpec struct {
+	// Resources defines requests and limits resources of VMI
+	Resources *kubevirtv1.ResourceRequirements `json:"resources"`
 	// SourceURL is the HTTP URL of the source image imported by CDI.
 	SourceURL string `json:"sourceURL,omitempty"`
 	// StorageClassName is the name which CDI uses to in order to create claims.
+	// +optional
 	StorageClassName string `json:"storageClassName,omitempty"`
 	// PVCSize is the size of the PersistentVolumeClaim that is created during the image import by CDI.
-	PVCSize string `json:"pvcSize,omitempty"`
-	// CPUs is the number of CPUs requested by the VM.
-	CPUs string `json:"cpus,omitempty"`
-	// Memory is the amount of memory requested by the VM.
-	Memory string `json:"memory,omitempty"`
+	// +optional
+	PVCSize *resource.Quantity `json:"pvcSize,omitempty"`
```

Comment:
Since this is not really optional, could we use `resource.Quantity` (not a pointer)?